ui/src/side_nav/containers/SideNav.js

```diff
@@ -25,6 +25,37 @@ const SideNav = React.createClass({
     }).isRequired,
     isHidden: bool.isRequired,
     logoutLink: string,
+    customLinks: arrayOf(
+      shape({
+        name: string.isRequired,
+        url: string.isRequired,
+      })
+    ),
+  },
+
+  renderUserMenuBlockWithCustomLinks(customLinks, logoutLink) {
+    return [<NavHeader key={0} title="User" />].concat(
```

Comment:
Have you looked into using the array spread operator to simplify these concats?

i.e., 

```javascript
[<NavHeader key={0} title="User" />, ...customLinks.map(), <NavListItem>Logout</NavListItem>]
```